data_structures/linked_list/doubly_linked_list.c

```diff
@@ -9,265 +9,221 @@
- * @param   pos     a position into the list for value insertion
- * @returns list    the input list with a node more or the same list
+ * @brief Insertion by position into the list function
+ * @param list head of a doubly linked List
+ * @param value a value to be inserted into the list
+ * @param pos position to insert the new node at
+ * @returns list the input list with the added node or the same list
  */
-List *insert(List *list, double value, int pos)
-{
-    // list NULL case
-    if (list == NULL)
-    {
+List *insert(List *list, double value, int pos) {
+    if (list == NULL) {
         list = create(value);
         return list;
     }
 
     // position existing case
-    if (pos > 0)
-    {
+    if (pos > 0) {
         List *cpy = list, *tmp = cpy;
         int flag = 1, index = 1, size = 0;
 
-        while (tmp != NULL)
-        {
+        while (tmp != NULL) {
             size++;
             tmp = tmp->next;
         }
 
         // first position case
-        if (pos == 1)
-        {
+        if (pos == 1) {
             List *new_node = create(value);
+            
             new_node->next = cpy;
             cpy->prev = new_node;
             list = new_node;
+
             return list;
         }
 
         // position existing in list range case
-        if (size + 2 > pos)
-        {
-            while (cpy->next != NULL && index < pos)
-            {
+        if (size + 2 > pos) {
+            while (cpy->next != NULL && index < pos) {
                 flag++;
                 index++;
                 cpy = cpy->next;
             }
 
-            List *new_node = (List *)malloc(sizeof(List));
+            List *new_node = (List *) malloc(sizeof(List));
             new_node->value = value;
 
-            // position into list with no poiting for NULL
-            if (flag == pos)
-            {
+            // position into list with no positioning for NULL
+            if (flag == pos) {
                 cpy->prev->next = new_node;
                 new_node->next = cpy;
                 new_node->prev = cpy->prev;
                 cpy->prev = new_node;
             }
 
             // last position case
-            if (flag < pos)
-            {
+            if (flag < pos) {
                 new_node->next = cpy->next;
                 new_node->prev = cpy;
                 cpy->next = new_node;
             }
         }
+
         return list;
     }
 }
 
 /**
- * @brief   Deletion by position into the list function
- * @param   list    a doubly linked List
- * @param   pos     a position into the list for value Deletion
- * @returns list    the input list with deleted values or the same list
+ * @brief Deletion of node at certain position
+ * @param list head of a doubly linked List
+ * @param pos position of node to be deleted
+ * @returns list the updated list
  */
-List *delete(List *list, int pos)
-{
-    // list NULL case
-    if (list == NULL)
+List *delete(List *list, int pos) {
```

Comment:
You'll have to rename this function as C++ will take it as you are using the `delete` built-in function.
I suggest renaming it to `delete_list`. Tested that and works perfectly. 👍
```suggestion
List *delete_list(List *list, int pos) {
```